### PR TITLE
SAT: declare `bypass_reason` in `acceptance-test-config.yml`

### PR DESCRIPTION
**Pull request by alafanechere**
## What
1/2 PR to complete this new feature: https://github.com/airbytehq/airbyte/issues/18219
We want to allow connector maintainers to express a `bypass_reason` on tests they might bypass.

## How
1. Change the `Config` model (used to parse `acceptance-test-config.yml` file from connectors) 
**from**: 
```yaml
connector_image: airbyte/source-pokeapi:dev
tests:
  connection:
    - config_path: "integration_tests/config.json"
      status: "succeed"
  discovery:
    - config_path: "integration_tests/config.json"
  basic_read:
    - config_path: "integration_tests/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
  full_refresh:
    - config_path: "integration_tests/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
```
**to**: 
```yaml
connector_image: airbyte/source-pokeapi:dev
acceptance_tests:
  spec:
    bypass_reason: "TODO: activate this test!"
  connection:
    tests:
      - config_path: "integration_tests/config.json"
        status: "succeed"
  discovery:
    tests:
      - config_path: "integration_tests/config.json"
  basic_read:
    tests:
      - config_path: "integration_tests/config.json"
        configured_catalog_path: "integration_tests/configured_catalog.json"
  full_refresh:
    tests:
      - config_path: "integration_tests/config.json"
        configured_catalog_path: "integration_tests/configured_catalog.json"
```

2. Avoid breaking changes by writing a config migration function to convert legacy config to the new config format using a pydantic root validator.
3. Update `pytest_generate_tests` + `pytest_collection_modifyitems` to handle the new config format 
4. Update `acceptance-test-config.yml.hbs` template in connector generator to reflect this change
5. Update `source-pokeapi` `acceptance-test-config.yml` as an example.

## Recommended reading order
1. https://github.com/airbytehq/airbyte/blob/e9e02a39d6f3d2d8d7d69866ac199afab1bc74b1/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/config.py#L163
2. https://github.com/airbytehq/airbyte/blob/e9e02a39d6f3d2d8d7d69866ac199afab1bc74b1/airbyte-integrations/bases/source-acceptance-test/unit_tests/test_config.py#L12
3. https://github.com/airbytehq/airbyte/blob/e9e02a39d6f3d2d8d7d69866ac199afab1bc74b1/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/plugin.py#L37
4. https://github.com/airbytehq/airbyte/blob/e9e02a39d6f3d2d8d7d69866ac199afab1bc74b1/airbyte-integrations/connector-templates/source_acceptance_test_files/acceptance-test-config.yml.hbs#L4
6. https://github.com/airbytehq/airbyte/blob/e9e02a39d6f3d2d8d7d69866ac199afab1bc74b1/airbyte-integrations/connectors/source-pokeapi/acceptance-test-config.yml#L3

## 🚨 User Impact 🚨
**This PR only affects the test configuration structure. It's not changing the acceptance tests. No connector update is required.**

### CODE REVIEW

**Comment by alafanechere** [comment]
☝️ Test on a migrated config (`source-pokeapi`) + 2 GA connectors without migrated config (`source-instagram`, `source-google-sheets`)

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/config.py [hunk 1]
```diff
@@ -133,22 +140,86 @@ class IncrementalConfig(BaseConfig):
...
+        """Root level validator executed 'pre' field validation to migrate a legacy config to the current structure.
+
+        Args:
+            values (dict): The raw configuration.
+
+        Returns:
+            dict: The migrated configuration if needed.
+        """
+        if ALLOW_LEGACY_CONFIG and cls.is_legacy(values):
+            logging.warn("The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format.")
```
Comment (evantahler) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/config.py:222]: Any docs for how to migrate to the new format?

**Review by evantahler — COMMENTED** [review]
Since we are versioning the interface to the SAT secretly here (is_legacy), we should also:
1. Add some docs about how to migrate the test YAMLs to the new format
2. add an issue in the icebox to remove the legacy code entirely at some point in the future

**Comment by alafanechere** [comment]
@evantahler thanks for the suggestions:
* I updated the reference doc
* I wrote a script to migrate a configuration
* I explained in the README how to run this script

**Review by evantahler — APPROVED** [review]

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**